c_api/src/taichi_core_impl.cpp

```diff
@@ -238,7 +238,7 @@ TiTexture ti_allocate_texture(TiRuntime runtime,
 
   switch ((taichi::lang::ImageDimension)allocate_info->dimension) {
 #define PER_IMAGE_DIMENSION(x) case taichi::lang::ImageDimension::x:
-#include "taichi/inc/image_dimension.inc.h"
```

Comment:
Since they are all guarded by a macro, should be easy to detect and process in the python script? I think keeping them seperate will end up being a lot of files for all kinds of different enums

Reply:
I see, I can somehow workaround that.